addon/validations/factory.js

```diff
@@ -564,7 +563,7 @@ function getCPDependentKeysFor(attribute, model, validations) {
   let dependentKeys = validations.map((validation) => {
     let { options } = validation;
     let type = validation._type;
-    let Validator = type === 'function' ? BaseValidator : lookupValidator(owner, type);
+    let Validator = type === 'function' ? BaseValidator : lookupValidator(owner, type).class;
```

Comment:
I understand. In this case you are not getting the factory to create an instance of that class, you want to read some information from the class itself.